akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -61,7 +64,21 @@ private[akka] class DaemonMsgCreateSerializer(val system: ExtendedActorSystem) e
           .setClazz(props.clazz.getName)
           .setDeploy(deployProto(props.deploy))
         props.args map serialize foreach builder.addArgs
-        props.args map (a ⇒ if (a == null) "null" else a.getClass.getName) foreach builder.addClasses
+        props.args.map { a ⇒
+          val argClassName =
+            if (a == null) "null"
+            else {
+              val className = a.getClass.getName
+              if (scala212OrLater && a.getClass.isInstanceOf[Serializable] && className.contains("$Lambda$")) {
+                // The serialization of the parameters is based on passing class name instead of
```

Comment:
When we serialize the Props we have one list with the serialized parameter values and another list with corresponding class names. When deserializing we load the classes from these class names to find out what serializer to use.

The problem is that the class name for a lambda can't be loaded from a name like that. lambda objects created through LambdaMetaFactory.

Instead of passing the lamda class name I pass `"java.io.Serializable"`.

This way of using serialization is wrong, but I can't change, as explained in comment above.